source/extensions/common/wasm/v8/v8.cc

```diff
@@ -554,6 +617,26 @@ void V8::registerHostFunctionImpl(absl::string_view moduleName, absl::string_vie
   host_functions_.emplace(absl::StrCat(moduleName, ".", functionName), std::move(func));
 }
 
+template <typename... Args>
+void V8::registerHostFunctionImpl(absl::string_view moduleName, absl::string_view functionName,
+                                  Word (*function)(void*, Args...)) {
+  ENVOY_LOG(trace, "[wasm] registerHostFunction(\"{}.{}\")", moduleName, functionName);
+  auto type = wasm::FuncType::make(convertArgsTupleToValTypes<std::tuple<Args...>>(),
+                                   convertArgsTupleToValTypes<std::tuple<Word>>());
+  auto func = wasm::Func::make(
+      store_.get(), type.get(),
+      [](void* data, const wasm::Val params[], wasm::Val results[]) -> wasm::own<wasm::Trap*> {
+        auto args_tuple = convertValTypesToArgsTuple<std::tuple<Args...>>(params);
+        auto args = std::tuple_cat(std::make_tuple(current_context_), args_tuple);
+        auto function = reinterpret_cast<Word (*)(void*, Args...)>(data);
+        Word rvalue = absl::apply(function, args);
+        results[0] = wasm::Val::make(static_cast<uint32_t>(rvalue.u64));
```

Comment:
Similarly, you could change this line to `results[0] = makeVal(rvalue);` in the the version above, and then we wouldn't need to have a specialized version for `Word`.

Reply:
Done and the next one down as well.